src/dawn/CodeGen/CodeGen.cpp

```diff
@@ -8,6 +8,20 @@ size_t CodeGen::getVerticalTmpHaloSize(Stencil const& stencil) {
   return (tmpInterval != nullptr) ? std::max(tmpInterval->overEnd(), tmpInterval->belowBegin()) : 0;
 }
 
+size_t CodeGen::getVerticalTmpHaloSizeForMultipleStencils(
+    const std::vector<std::shared_ptr<Stencil>>& stencils) const {
+  std::shared_ptr<Interval> fullIntervals = stencils[0]->getEnclosingIntervalTemporaries();
```

Comment:
There are some issues, stencils can be an empty vector, and stencils[0] will segfault. 
I would suggest to replicate the same algorithm we have in the Stencil.cpp
But also I realized that we are abusing here shared_ptr, we just want to have an output that could have a non initialized state. I think we should use boost::optional, as I was doing in the PR #74 
But I would suggest you to fix this function as we do it in Stencil.cpp and later we refactor all the getEnclosingIntervalTemporaries using a boost::optional if you agree

Also would be good to unittest this